### ISSUE TITLE
GLIBC errors when importing healpy

### ISSUE THREAD

**Issue by woodmd** [OpenAstronomy/conda-channel#8]
I've been experimenting with the healpy conda package but I'm encountering some problems when installing heapy on older Linux distributions -- specifically SL6 and RHEL6.  When I try to import healpy I get an error about the version of glibc:

```
(fermi-env)[vagrant@sl65-minimal ~]$ python -c "import healpy"
/home/vagrant/miniconda/envs/fermi-env/lib/python2.7/site-packages/healpy/pixelfunc.py:105: UserWarning: Warning: cannot import _healpy_pixel_lib module
  warnings.warn('Warning: cannot import _healpy_pixel_lib module')
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/vagrant/miniconda/envs/fermi-env/lib/python2.7/site-packages/healpy/__init__.py", line 42, in <module>
    from .sphtfunc import (anafast, map2alm,
  File "/home/vagrant/miniconda/envs/fermi-env/lib/python2.7/site-packages/healpy/sphtfunc.py", line 31, in <module>
    from . import _healpy_sph_transform_lib as sphtlib
ImportError: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by /home/vagrant/miniconda/envs/fermi-env/lib/python2.7/site-packages/healpy/_healpy_sph_transform_lib.so)
```

The problem seems to be that SL6/RHEL6 come with glibc 2.12 while the healpy conda package was compiled against glibc 2.14.  Is there a workaround for this?

**Comment by Cadair** [comment]
We might have to take a leaf out of the conda-forge book and build on a different CI server.

**Comment by woodmd** [comment]
Are there are any plans to revisit this?  I realize that RHEL6 is a pretty old distribution but many institutions still use it in their computing cluster environments (my own for instance).  I presume that this issue will come up whenever you have compiled extensions so having a solution for this would generally be useful for packages beyond healpy and also going forward as CI services migrate to newer versions of Linux.

**Comment by mwcraig** [comment]
> Are there are any plans to revisit this? 

@woodmd -- yes. 

TL;DR: will take at least a few days to get it set up, but the need is real and not unique to you.

I think the easiest fix (from our POV) is to get healply building on conda-forge, then copy that build to this channel. conda-forge already has the infrastructure set up for doing builds on older build machines. Adding that infrastructure to this repo (circle-ci with docker images) seems much more time consuming than simply using conda-forge and making the PR to the appropriate repo for allowing cross-channel package copying.

**Comment by mwcraig** [comment]
@woodmd -- can you please try this version of healpy and let me know either way if it works for you:

```
conda install -c conda-forge healpy
```

If it works, I'll copy that build into this channel (instead of us building it directly).

**Comment by woodmd** [comment]
Yes the conda-forge version works on both SL6 and RHEL6.  Thanks!

Just out of curiosity if healpy is available through conda-forge what is the motivation for providing it through openastronomy as well?  Would there be any difference between the versions I would get from the conda-forge vs. openastronomy channel?

**Comment by mwcraig** [comment]
At this point there isn't much motivation, unless openastronomy has other packages you need that are not in conda-forge yet.

As conda-forge continues to grow I see things like openastronomy the astropy channel as collections of related packages that are actually built in conda-forge (their infrastructure is really nice).

**Comment by woodmd** [comment]
Did you finish importing the conda-forge build into openastronomy?  I just tried installing healpy from openastronomy but I'm still getting the GLIBC error.

**Comment by mwcraig** [comment]
Should get to it Sunday.

**Comment by mwcraig** [comment]
@woodmd -- the correct packages should be available now (still haven't automated that, but that is a separate issue...).

I'm closing this, but if you run into a problem please do reopen, or open a separate issue.
